### PR TITLE
Created new unit test for issue 509 to show the problem appears to be resolved

### LINKED ISSUE #509: Returned driver instance from switchTo().window() have  0  ImplicityWait timeout
selenium 2.45.0
firefox 37.0.1

Setting up my implicity wait by
<code>
driver.manage().timeouts().implicitlyWait(15, TimeUnit.SECONDS);
</code>

then switching to another window and storing returned driver instance to a new variable
<code>WebDriver windowDriver = driver.switchTo().window(handleHash);</code>

When trying to interact with element in another window with windowDriver variable (which we got from switchTo().window()  method) I'm getting NoSuchElement exception with 50-200 miliseconds wait.
<code>windowDriver.findElement()</code>

Problem solved when I put once again implicityWait but directly to new variable windowDriver.
<code>windowDriver.manage().timeouts().implicitlyWait(15, TimeUnit.SECONDS);</code>

P.S. sessionID and objectID in debugging are the same.

### PR DESCRIPTION
**Pull request by gregorykleiner**
#509 new unit to prove implicit wait issue is now resolved

### CODE REVIEW

File: java/client/test/org/openqa/selenium/ImplicitWaitTest.java [hunk 1]
```diff
@@ -154,4 +155,28 @@ public void testShouldImplicitlyWaitForAnElementToBeVisibleBeforeInteracting() {
       fail("Element should have been visible");
     }
   }
+
+
+  @Test
+  @JavascriptEnabled
+  @Ignore({IE, PHANTOMJS, SAFARI, MARIONETTE})
+  public void testShouldRetainImplicitlyWaitFromTheReturnedWebDriverOfFrameSwitchTo() {
+    driver.manage().timeouts().implicitlyWait(3, SECONDS);
```
Comment (lukeis) [inline java/client/test/org/openqa/selenium/ImplicitWaitTest.java]: would rather do a smaller timeout, 1 second should be sufficient

Comment (gregorykleiner) [inline java/client/test/org/openqa/selenium/ImplicitWaitTest.java]: lukeis,
That is fine.  We did 3 seconds to make sure there was actually wait after the findElements failed.  I will update to 1 second

**Comment by santiycr** [comment]
@gregorykleiner thanks for your contribution!! If you're interested in a Sauce swag box, can you share your address and shirt size with me to sso @ saucelabs com please?

**Comment by lukeis** [comment]
merged as bd7d47f501324a733b9fd07171600e17b6ff2116

**Comment by lukeis** [comment]
@gregorykleiner whoops, forgot to ask about the [CLA](https://docs.google.com/forms/d/11Z8LoYpTGUIwCegifVH1YtL9smxVDNk-fOykUZTAWhE/viewform?hl=en_US&formkey=dFFjXzBzM1VwekFlOWFWMjFFRjJMRFE6MQ#gid=0), can you please sign it :)

**Pull request closed without merging**